sdk/include/reactos/shellutils.h

```diff
@@ -227,25 +227,25 @@ class CComDebugObject : public Base
 #endif
 
 template<class T>
-void ReleaseCComPtrExpectZero(CComPtr<T>& cptr, BOOL forceRelease = FALSE)
+ULONG ReleaseCComPtrExpectZeroHelper(CComPtr<T>& cptr, BOOL forceRelease = FALSE)
 {
+    ULONG r = 0;
     if (cptr.p != NULL)
     {
         T *raw = cptr.Detach();
-        int nrc = raw->Release();
-        if (nrc > 0)
+        int nrc = r = raw->Release();
+        while (nrc > 0 && forceRelease)
         {
-            DbgPrint("WARNING: Unexpected RefCount > 0 (%d)!\n", nrc);
-            if (forceRelease)
-            {
-                while (nrc > 0)
-                {
-                    nrc = raw->Release();
-                }
-            }
+            nrc = raw->Release();
         }
     }
+    return r;
 }
+#define ReleaseCComPtrExpectZero(...) \
```

Comment:
I would strongly suggest the usage of the `do { ... } while (0)` construct here, in order for the macro to be usable as a function.

Reply:
I don't know what you mean by usable as a function, it is already usable as a function since clearly other code calls it as a function.